### DISCUSSION TITLE
Compiling issue

### DISCUSSION THREAD

**Discussion by shrivrat0809** [Cyfrin/foundry-full-course-cu#261] [Q&A] [answered]
my code is not compiling in remix ide it is showing spdx license not provided.
how can i compil my code because patrick is doing same and he is able to compile his code.
please help me !!!

**Comment by ghost** [comment, marked as answer]
Hello @shrivrat0809 try adding on the first line of the file the following:

```
// SPDX-License-Identifier: MIT
```

If the issue persist, share the file following these [guidelines](https://github.com/Cyfrin/foundry-full-course-f23/blob/main/how-to-ask-a-question.md), with emphasis on point 5.

**Reply by malommey** [reply]
check your version make sure you are using compiler version ^0.8.18 if it doesn’t work type pragma with tab let remix ide complete the whole statement for you including the SPDX license.. try it

**Reply by cromewar** [reply]
@shrivrat0809 please confirm any of this responses helped you to solve the problem.
